workers/unity/Assets/Playground/Scripts/Metrics/MetricSendSystem.cs

```diff
@@ -37,52 +47,37 @@ protected override void OnUpdate()
             }
 
             timeElapsedSinceUpdate += Time.deltaTime;
-            
+
             AddFpsSample();
             if (timeElapsedSinceUpdate >= TimeBetweenMetricUpdatesSecs)
             {
                 timeElapsedSinceUpdate = 0;
-                var fps = CalculateFps();
-                var load = DefaultLoadCalculation(fps);
-                var metrics = new Improbable.Worker.Metrics
-                {
-                    Load = load
-                };
-                connection.SendMetrics(metrics);
-            }
-        }
 
-        private static float DefaultLoadCalculation(float fps)
-        {
-            float targetFps = Application.targetFrameRate;
+                var dynamicFps = CalculateFps();
+                Metrics.GaugeMetrics["Dynamic.FPS"] = dynamicFps;
+                Metrics.GaugeMetrics["Unity used heap size"] = GC.GetTotalMemory(false);
+                Metrics.Load = CalculateLoad(dynamicFps);
 
-            if (targetFps == -1)
-            {
-                targetFps = DefaultTargetFrameRate;
+                connection.SendMetrics(Metrics);
             }
+        }
 
-            return Mathf.Max(0.0f, (targetFps - fps) / (0.5f * targetFps));
+        private float CalculateLoad(float dynamicFps)
+        {
+            return Mathf.Max(0.0f, 0.5f * TargetFps / dynamicFps);
         }
 
         private void AddFpsSample()
         {
-            if (fpsMeasurements.Count == MaxFpsSamples)
-            {
-                fpsMeasurements.Dequeue();
-            }
-
-            fpsMeasurements.Enqueue(1.0f / Time.deltaTime);
+            totalFpsCount += 1.0f / Time.deltaTime;
+            fpsMeasurements++;
```

Comment:
If I understand correctly, this is number of frames, right? Can we name it frameCount to be more clear?

Reply:
renamed 👍